test/CMakeLists.txt

```diff
@@ -1062,31 +1063,50 @@ ecbuild_add_test( TARGET  test_iodaconv_atms_nc4
                           -i testinput/SNDR_SNPP_ATMS_SAMPLE.nc4
                           -o testrun/2021120600_atms_sdr.nc4" 2021120600_atms_sdr.nc4 ${IODA_CONV_COMP_TOL_ZERO} )
 
+if( iodaconv_eccodes_ENABLED )
```

Comment:
@rhoneyager   I just added this ```iodaconv_eccodes_ENABLED``` since we should probably protect against the usage of scripts contributed by Ben Ruston and myself until eccodes can be installed on as many platforms as possible.  Meanwhile, I don't know how to modify the cmake codes to create this new variable.  So I am seeking  your help to enable this macro please.

Reply:
The eccodes find_package script should report whether it is available of not. Checking that variable should be fine. It's something like ```eccodes_FOUND```.

```find_package(eccodes .......``` should also be added in the top-level CMakeLists.txt file.

Reply:
Pinging @climbfuji here since we just added eccodes Python in addition to eccodes C, and we need to develop a find_package script in jedi-cmake for this.

Reply:
Duly notes, thanks for the heads up.

Reply:
@gthompsnJCSDA You could do something similar to what is done for pygrib in the top level CMakeLists.txt file. The following suggestion is code to be placed in the top level CMakeLists.txt file.

```
option(USE_ECCODES "Use Python eccodes package" ON)
if ( USE_ECCODES )
    message(STATUS "Adding python tests that import eccodes")
    set( iodaconv_eccodes_ENABLED True)
else()
    message(STATUS "Skipping python tests that import eccodes")
    set( iodaconv_eccodes_ENABLED False)
endif()
```

What this will do is set USE_ECCODES to ON by default. If you need to shut these off, then you add `-DUSE_ECCODES=OFF` to the ecbuild command line.

Reply:
Sorry, when writing my review I didn't see @rhoneyager's response. @rhoneyager has a good point about using find_package. I had forgotten that eccodes comes in two pieces, a compiled library (of which the find_package is needed to check for its existence) and a python package (of which the option is needed to get the hint from the user). I'm not sure if there is a way to automatically detect the existence of the python eccodes package. Perhaps cmake has a way to do that.

For now you could add the find package and change the if statement to something like:
```
find_package( eccodes QUIET )
option(USE_ECCODES "Use Python eccodes package" ON)
if ( eccodes_FOUND AND USE_ECCODES )
...
```

Reply:
After today's EMC IODA Hour discussion, I think we should make the default for USE_ECCODES be OFF instead of ON. Not all operational centers will use eccodes, and therefore eccodes should be an optional part of jedi-stack (as well as the eccodes python package being optional).